File: JavaScript/ExampleTests/calculator_glue.js
import { AdderTyped } from "./common/index.js";
import { Calculator } from "./production/index.js";

export class CalculatorGlue {
  static DNC_STRING = "?DNC?";

  constructor() { this.calc = new Calculator(); }

  examplesCalculationAddTwoNumbers(values) {
    values.forEach((value) => {
      const typed = AdderTyped.fromStringObj(value);
      expect(this.calc.add(typed.number1, typed.number2)).toBe(typed.result);
    });
  }
}

File: Python/ExampleTests/calculator_glue.py
from common import *
from production import Calculator


class CalculatorGlue:
    DNC_STRING = '?DNC?'

    def examples_Calculation_AddTwoNumbers(self, values: list):
        calc = Calculator()
        for value in values:
            typed = AdderTyped.from_string_obj(value)
            actual = calc.add(typed.number1, typed.number2)
            assert typed.result == actual, \
                f'Add {typed.number1} + {typed.number2}'
